### PR TITLE
POC of a new design system approach, fixed bugs

### PR DESCRIPTION
**Pull request by dmitryrazinkov**
* POC of a new design system approach:
  - responsive grid and typography values
  - added breakpoints mixin
  - refactored a part of existing components (cats page)
* Fixed store usage, improved typings

### CODE REVIEW

File: src/components/LoginForm.vue [hunk 1]
```diff
@@ -96,39 +92,21 @@ export default defineComponent({
...
-    padding-bottom: $gridLarge;
-    align-self: center;
-    &__icon {
-      height: 64px;
-      width: 48px;
-      margin-right: $gridNormal;
-    }
+  display: grid;
+  background-color: var(--background-light-color);
+  padding: var(--space-xxxxxl) var(--space-xxxxl);
```
Comment (Yar-s) [inline src/components/LoginForm.vue:97]: I find it difficult to count `x`'s at glance.

File: src/components/VButton.vue [hunk 2]
```diff
@@ -17,19 +17,31 @@ export default defineComponent({
 @import "src/scss/typography";
 
 .button {
-  background: $primaryColor;
+  background: var(--primary-color);
   border-radius: 4px;
```
Comment (Yar-s) [inline src/components/VButton.vue:21]: Let's use constants for `border-radius` and `font-weight`?

File: src/components/VButton.vue [hunk 3]
```diff
@@ -17,19 +17,31 @@ export default defineComponent({
...
   border: none;
-  text-transform: uppercase;
-  display: inline-block;
+  display: grid;
   cursor: pointer;
   text-align: center;
   letter-spacing: 0.05em;
 
+  &--icon {
+    padding: var(--space-sm) var(--space-sm);
```
Comment (Yar-s) [inline src/components/VButton.vue:34]:
```suggestion
    padding: var(--space-sm);
```

File: src/components/VHeader.vue [hunk 4]
```diff
@@ -1,14 +1,15 @@
...
-        {{ i18n.t("cats") }}
-      </span>
+      <img class="header__logo__icon" src="/logo.svg" />
     </div>
-    <v-button @click="$emit('logout', $event)" class="header__logout">
-      {{ i18n.t("logout") }}
+    <v-button
+      @click="$emit('logout', $event)"
+      v-if="isAuthenticated"
+      class="button--wrapper"
```
Comment (Yar-s) [inline src/components/VHeader.vue:9]: Is it ok to merge outer and inner classes?

File: src/components/VInput.vue [hunk 5]
```diff
@@ -1,15 +1,20 @@
...
+        :name="name"
+        :class="{ 'input--not-empty': !!modelValue }"
+        @input="onInput($event)"
+      />
+      <label class="input-label" :for="name">{{ label }}</label>
+    </div>
+    <div class="error-container">
+      {{ this.errorMessage || serverError }}
+    </div>
+  </div>
```
Comment (Yar-s) [inline src/components/VInput.vue:17]: Do we need this outer `<div>`?

File: src/components/VInput.vue [hunk 6]
```diff
@@ -1,15 +1,20 @@
...
-  </label>
+  <div>
+    <div class="input-container">
+      <input
+        class="input"
+        :value="modelValue"
+        :type="type"
+        :name="name"
+        :class="{ 'input--not-empty': !!modelValue }"
+        @input="onInput($event)"
```
Comment (Yar-s) [inline src/components/VInput.vue:10]: Not sure, maybe it's better to use a `v-model` here and `handleChange` in a `watch`'er?
I'm new to Vue yet, so it looks to me like a step back from its proposed way of doing things...

File: src/components/VInput.vue [hunk 7]
```diff
@@ -65,18 +70,49 @@ export default defineComponent({
...
-  color: $textColor;
-  @include text-normal;
+  width: 100%;
+  box-sizing: border-box;
+  outline: none;
+  background: none;
+  border-style: solid;
+  border-width: var(--border-xs);
+  border-color: var(--border-color);
+  border-radius: 4px;
```
Comment (Yar-s) [inline src/components/VInput.vue:86]: Looks like there's a constant for this

File: src/scss/breakpoints.scss [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+        md: 64rem, // 1024px
+        lg: 80rem, // 1280px
+        xl: 90rem  // 1440px
+);
+
+// max-width media query, can be used as
+// @include breakpoint(sm, "(orientation: landscape)") {
+// }
+// the second param is additional logic
+@mixin breakpoint($breakpoint, $logic: false) {
```
Comment (Yar-s) [inline src/scss/breakpoints.scss:13]: What about renaming `breakpoint` to `bp`?

**Review by Yar-s — COMMENTED** [review]
Could you clarify, what's the purpose of moving from SASS constants to CSS variables?

File: src/components/VInput.vue (hunk 6 above)
Comment (Yar-s) [reply src/components/VInput.vue:10]: Oh, I see. It's done to use `v-model` on `<v-input>`

File: src/components/LoginForm.vue (hunk 1 above)
Comment (dmitryrazinkov) [reply src/components/LoginForm.vue:97]: Agree, we will probably find other namings, it's the first version for now.

File: src/components/VButton.vue (hunk 2 above)
Comment (dmitryrazinkov) [reply src/components/VButton.vue:21]: Yes, I'll add it a bit later

File: src/components/VHeader.vue (hunk 4 above)
Comment (dmitryrazinkov) [reply src/components/VHeader.vue:9]: Yes, it's one of the features of Vue, attributes that you apply to a component automatically applied to a root node. So you can significantly reduce the number of additional props.

File: src/components/VInput.vue (hunk 5 above)
Comment (dmitryrazinkov) [reply src/components/VInput.vue:17]: yes, in vue components should have only one root node

File: src/components/VInput.vue (hunk 6 above)
Comment (dmitryrazinkov) [reply src/components/VInput.vue:10]: yes =)

File: src/components/VInput.vue (hunk 7 above)
Comment (dmitryrazinkov) [reply src/components/VInput.vue:86]: yeah, will do it

File: src/scss/breakpoints.scss (hunk 8 above)
Comment (dmitryrazinkov) [reply src/scss/breakpoints.scss:13]: hmm... I think it's better to leave as it's to reduce any concerns

**Comment by dmitryrazinkov** [comment]
> Could you clarify, what's the purpose of moving from SASS constants to CSS variables?

Yes, actually there is a bunch of reasons. As for grid and typography, it allows us to make them responsive, so they adjust to screen size. As for colors, it allows us to replace them for different themes, which makes it really easy to support different theming. 

Actually, css constants are more powerful as they are nested and can be overridden, accessed in JS, and so on. We looked at the best design systems and were inspired by these ideas from them. Anyway, this approach is still under work and we'll define it somehow once we are ready.

File: src/components/VButton.vue (hunk 2 above)
Comment (dmitryrazinkov) [reply src/components/VButton.vue]: done

File: src/components/VInput.vue (hunk 7 above)
Comment (dmitryrazinkov) [reply src/components/VInput.vue]: done

**Review by Yar-s — APPROVED** [review]

**Pull request merged**